lib/my_array.rb

```diff
@@ -0,0 +1,50 @@
+class MyArray < Array
+  def initialize(array)
+    @array = array
+  end
+
+  def inject(arg1 = nil, arg2 = nil, &block)
+    if arg1.is_a? Symbol
+      return inject_symbol(arg1)
+    elsif arg2.is_a? Symbol
+      return inject_symbol_with_initial(arg1, arg2)
+    elsif arg1.nil?
+      return inject_block(&block)
+    else
+      return inject_block_with_initial(arg1, &block)
+    end
+  end
+
+  private
+
+  attr_reader :array
```

Comment:
Boo-urns: these weird me out this far down a class. `attr_reader` should really only declare public interfaces